packages/typed-message/base/utils/index.ts

```diff
@@ -26,6 +28,22 @@ export function isSerializableTypedMessage(x: TypedMessage): x is SerializableTy
 export function isTypedMessageEqual(message1: TypedMessage, message2: TypedMessage): boolean {
     if (message1.type !== message2.type) return false
     if (message1.meta !== message2.meta) return false
-    // perform a deep equal
-    return isEqual(message1, message2)
+    return renderMessage(message1) === renderMessage(message2)
```

Comment:
This post https://twitter.com/Nate_FTXUS/status/1499219037552668674 , for example, flatten message is not equal to transformed message


https://github.com/DimensionDev/Maskbook/blob/801c3e8659b89ffad6d0648614153a88cc03daac/packages/mask/src/components/InjectedComponents/PostReplacer.tsx#L63

Reply:
hmm I think it should not equal. For example, `[[1, 2, 3]].flat()` is not the same as `[1, 2, 3]`